### PR TITLE
ADAPT-2932 Add spacing options and full viewport style

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- Add spacing options and full viewport style

# Review By (Date)
- July 7

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to...
3. Verify...

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- [ADAPT-2932](https://stanfordits.atlassian.net/browse/ADAPT-2932)

### CODE REVIEW

File: src/components/media/simpleImage.js [hunk 1]
```diff
@@ -1,27 +1,72 @@
...
+  containerAlign,
+} from "../../utilities/dataSource";
 
 const SimpleImage = ({
-  blok: { image: { filename, alt } = {}, imageWidth, caption, captionAlign },
+  blok: {
+    image: { filename, alt } = {},
+    imageWidth,
+    imageFocus,
+    isBanner,
```
Comment (rebeccahongsf) [inline src/components/media/simpleImage.js:19]: @yvonnetangsu what are our thoughts on having a boolean value to have the full width/height banner? 🤔  That way we don't need to create a new edge-to-edge class and adjust the widthBox.

Comment (yvonnetangsu) [reply src/components/media/simpleImage.js:19]: @rebeccahongsf Nice idea - I like it! 👏🏼  😍

**Review by sherakama — APPROVED** [review]
This looks fine to me with @yvonnetangsu's approval.

File: src/components/media/simpleImage.js [hunk 2]
```diff
@@ -1,27 +1,72 @@
...
+  const spacingTopStyle = smallPaddingTop[spacingTop];
+  const spacingBottomStyle = smallPaddingBottom[spacingBottom];
+  const captionAlignment = textAlign[captionAlign ?? "left"];
+  const containerAlignment = containerAlign[captionAlign ?? "left"];
+  const imageFocusPosition = objectPosition[imageFocus ?? "center"];
+
+  let wrapperHeight = "";
+  let imageStyle = "";
+  if (imageWidth === "edge-to-edge" && isBanner) {
+    wrapperHeight = "su-relative su-h-screen su-w-full su-overflow-hidden";
```
Comment (yvonnetangsu) [inline src/components/media/simpleImage.js:36]: I like the boolean isBanner option, but on Giving, they usually use it as a narrower "band" of image (kind of like a poster without words). Could we copy over the heights of the full width image on Giving for different breakpoints? I think it was 500px or 600px tall for 2XL 😄

Comment (yvonnetangsu) [reply src/components/media/simpleImage.js:36]: Here's an example on this page (2nd image)
https://giving.stanford.edu/stories/academic-and-creative-journey/

Comment (yvonnetangsu) [reply src/components/media/simpleImage.js:36]: Also, could we add a boolean to add a `su-cc` centered container around the caption content, so when the image is full width, one can choose whether the caption start at the edge of of the parent container (eg, when it's placed inside the "main content" and the image is fit to 100% width of parent) or it start at the edge of the centered container (eg, when it's used as an edge to edge image on a page). Thanks 🙏🏼 😄

File: src/components/media/simpleImage.js [hunk 3]
```diff
@@ -1,27 +1,72 @@
...
+      <WidthBox
+        width={imageWidth}
+        className={dcnb(spacingTopStyle, spacingBottomStyle)}
+      >
+        <figure>
+          <div className={wrapperHeight}>
+            <img
+              src={filename}
+              alt={alt ?? ""}
+              className={dcnb(imageStyle, imageFocusPosition)}
```
Comment (yvonnetangsu) [inline src/components/media/simpleImage.js:51]: Let's add a `su-w-full` class to the image so its width will fill what it's supposed to occupy. I merged this into my branch and found that if the image is smaller than the width you choose, it stays at the smaller size which could be a little bit confusing 😃

Comment (yvonnetangsu) [reply src/components/media/simpleImage.js:51]: Screenshot showing what I was talking about:

Screen Shot 2021-07-06 at 1 26 31 AM

File: src/components/media/simpleImage.js (hunk 2 above)
Comment (rebeccahongsf) [reply src/components/media/simpleImage.js:38]: I added a `isCaptionCenter` boolean and accompanying help text "If Fit Parent Container is selected, caption can be adjusted to full width or center container aligned." which might need additional revision/massaging. 😄

**Comment by yvonnetangsu** [comment]
Hehe @rebeccahongsf 99% there! Thank you for the changes 😄 

The only issue left is with the Fit Parent Container (non-banner) option, eg, when on mobile breakpoints, the width of the image should fit the width of the device with the height being auto to retain the original image aspect ratio. Currently on mobile:
Screen Shot 2021-07-06 at 4 26 10 PM

File: src/components/media/simpleImage.js (hunk 2 above)
Comment (yvonnetangsu) [reply src/components/media/simpleImage.js:38]: I think the `su-h-screen` class is causing the issue. I think you can just remove it - we shouldn't need it for any options 😄

**Comment by rebeccahongsf** [comment]
> Hehe @rebeccahongsf 99% there! Thank you for the changes 😄
> 
> The only issue left is with the Fit Parent Container (non-banner) option, eg, when on mobile breakpoints, the width of the image should fit the width of the device with the height being auto to retain the original image aspect ratio. Currently on mobile:
> _screenshot_

Removing the `su-h-screen` seemed to resolve the mobile height issue! Please let me know if anything else appears out of place. Thanks @yvonnetangsu! 😄

**Review by yvonnetangsu — APPROVED** [review]
GTG - thank you for the revision! 🎉 😄

**Pull request merged**